src/08_example/05_tests.md

```diff
@@ -0,0 +1,53 @@
+# Testing the TCP Server
+Let's move on to testing our `handle_connection` function.
+First, we need a `TcpStream` to work with, but we don't want to make a real TCP connection in test code.
+We could work around this in a few ways.
+One strategy could be to refactor the code to be more modular,
+and only test that the correct responses are returned for the respective inputs.
+
+Another strategy is to connect to `localhost` on port 0.
```

Comment:
What's wrong with the port you used, 7878?

Reply:
7878 is fine, but I wrote this due to a preference that the test code should be able to run on any server, where other things might be using port 7878. I believe I also saw this used in the tests for `smol`. No strong preference here.